db/compaction/compaction_job.cc

```diff
@@ -1718,4 +1730,56 @@ void CompactionJob::LogCompaction() {
   }
 }
 
+class SstPartitionerFixedPrefix : public SstPartitioner {
+ public:
+  SstPartitionerFixedPrefix(size_t len) : len_(len) {}
+
+  virtual ~SstPartitionerFixedPrefix(){};
+
+  const char* Name() const override { return "SstPartitionerFixedPrefix"; }
+
+  bool ShouldPartition(const Slice& key) override {
+    if (last_key_.empty()) {
+      return false;
+    }
+    Slice key_fixed(key.data_, std::min(key.size_, len_));
+    return key_fixed.compare(last_key_) != 0;
```

Comment:
Shouldn't this be based on a Comparator rather than raw bytes?

Reply:
We compare prefix (number of bytes) for equality. I do not see much benefit to use comparator for this partitioner.